### ISSUE TITLE
Docutils mark-up for hyperlink reference ok for Sphinx HTML but not Sphinx LaTeX

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#5948]
Consider following `index.rst`

```rest
Welcome to FOO's documentation!
===============================

The latex_elements configuration setting
----------------------------------------

Its ``'sphinxsetup'`` key is described `separately <latexsphinxsetup_>`_.

.. _latexsphinxsetup:

\\'sphinxsetup\\' key
---------------------

The ``'sphinxsetup'``
provides a LaTeX-type customization interface.
```

When building via Docutils

```
rst2html.py index.rst index.html
rst2latex.py index.rst index.tex
```

no problem arises, and `pdflatex index.tex` does not complain about duplicated references. Indeed the `index.tex` contains a unique `\label{latexsphinxsetup}`.

But on `make latexpdf` we get a warning:

```console
Latexmk: Summary of warnings from last run of (pdf)latex:
  Latex found 1 multiply defined reference(s)
Latexmk: All targets (FOO.pdf) are up-to-date
```

and indeed the `_build/latex/FOO.tex` file contains two `\label{\detokenize{index:latexsphinxsetup}}` (at each  `\chapter` title).

On the other hand `make html` does not produce a problem, the `_build/latex/index.html` file contains only one ``<span id="latexsphinxsetup"></span>``.

Sphinx version: 1.8.4+
Python version: 3.6.6 (CPython)
Docutils version: 0.14 
Jinja2 version: 2.10

Relates: 7e851510eb09813ccef85632284fc2a6655bac07

My question is whether Docutils syntax is illegal for Sphinx or whether there is a problem with LaTeX writer ?

**Comment by tk0miya** [comment]
docutils is a specification and standard implementation of reST. So it is always valid, I think.
In addition, the example you shown is very natural to me. So I suppose this is a bug of LaTeX writer.

**Comment by tk0miya** [comment]
Note:
It seems LaTeX writer of docutils skips to render a target node having `refid` attributes.
https://github.com/docutils-mirror/docutils/blob/e88c5fb08d5cdfa8b4ac1020dd6f7177778d5990/docutils/writers/latex2e/__init__.py#L2833-L2838

This is a difference between Sphinx.

**Comment by tk0miya** [comment]
Fixed by #6006 :-)
